projects/wpdx/src/app/rehab-prio/adm-popup/adm-popup.component.ts

```diff
@@ -71,80 +71,53 @@ export class AdmPopupComponent implements OnChanges {
     return queries;                                                                                                                               
   }
 
+
+  addQuery(value, baseQuery, count, queries, attribute){
+    if (value){
+      baseQuery = `${baseQuery} and clean_adm${count}='${this.fq(value)}'`;
+      queries.push(`${baseQuery} group by ${attribute}`);
+    }
+    return baseQuery
+    
+  }
+
   getDataQuery(value, attribute){
-    const baseQuery = `select ${attribute}, count(${attribute}) from 
+    const values = [value.NAME_1, value.NAME_2, value.NAME_3, value.NAME_4];
+    let baseQuery = `select ${attribute}, count(${attribute}) from
     wpdx_plus`
     const queries: string[] = [];
-    if (value.NAME_0) {
-      queries.push(`${baseQuery}
-        where clean_country_name='${this.fq(value.NAME_0)}' group by ${attribute}`);
-    }
-    if (value.NAME_1) {
-      queries.push(`${baseQuery}
-        where clean_country_name='${this.fq(value.NAME_0)}' and clean_adm1='${this.fq(value.NAME_1)}' group by ${attribute}`);
+    if (value.NAME_0){
+      baseQuery = `${baseQuery} where clean_country_name='${this.fq(value.NAME_0)}'`;
+      queries.push(`${baseQuery} group by ${attribute}`);
     }
-    if (value.NAME_2) {
-      queries.push(`${baseQuery}
-        where clean_country_name='${this.fq(value.NAME_0)}' and clean_adm1='${this.fq(value.NAME_1)}' and 
-              clean_adm2='${this.fq(value.NAME_2)}' group by ${attribute}`);
-    }
-    if (value.NAME_3) {
-      queries.push(`${baseQuery}
-        where clean_country_name='${this.fq(value.NAME_0)}' and clean_adm1='${this.fq(value.NAME_1)}' and 
-              clean_adm2='${this.fq(value.NAME_2)}' and clean_adm3='${this.fq(value.NAME_3)}' group by ${attribute}`);
-    }
-
-    if (value.NAME_4) {
-      queries.push(`${baseQuery}
-        where clean_country_name='${this.fq(value.NAME_0)}' and clean_adm1='${this.fq(value.NAME_1)}' and 
-              clean_adm2='${this.fq(value.NAME_2)}' and clean_adm3='${this.fq(value.NAME_3)}' and clean_adm4='${this.fq(value.NAME_4)}' group by ${attribute}`);
+    for (let i = 0; i < values.length; i++){
+      baseQuery = this.addQuery(values[i], baseQuery, i+1, queries, attribute)
     }
     return queries;
 
   }
 
-  getManagementData(){
-    forkJoin(this.getDataQuery(this.popupProperties, 'management_clean').map(q => this.db.query(q))).subscribe(results => {
-      this.management_data = [];
+  getData(column_name){
+    forkJoin(this.getDataQuery(this.popupProperties, column_name).map(q => this.db.query(q))).subscribe(results => {
+      const data_array = [];
       for (let i = 0; i < results.length; i++){
```

Comment:
Better to use the `for (const result of results)` instead of the C-style `for (i = 0; i<len; i++)`.